crates/anvil-polkadot/src/api_server/mod.rs

```diff
@@ -32,13 +34,26 @@ pub fn spawn(
     let service = substrate_service.clone();
     let mut impersonation_manager = ImpersonationManager::default();
     impersonation_manager.set_auto_impersonate_account(config.enable_auto_impersonate);
+    let mut signers = config.signer_accounts.clone();
+    if let Some(genesis) = &config.genesis {
+        let genesis_signers = genesis
+            .alloc
+            .values()
+            .filter_map(|acc| acc.private_key)
+            .flat_map(|k| Keypair::from_secret_key(*k))
+            .collect::<Vec<_>>();
+        if !genesis_signers.is_empty() {
+            signers.extend(genesis_signers);
+        }
+    }
```

Comment:
super nit: probably something like this works too
```suggestion
   signers.extend(
            genesis
                .alloc
                .values()
                .filter_map(|acc| acc.private_key)
                .flat_map(|k| Keypair::from_secret_key(*k))
    );
```